src/chains/algorand.ts

```diff
@@ -0,0 +1,74 @@
+const algosdk = require('algosdk');
+const nacl = require('tweetnacl');
+
+function privateKeyToArray(privateKey: string) {
+    return Buffer.from(privateKey.substring(2,privateKey.length), 'hex').toJSON().data
+}
+
+function keyPairFromSecretKey(sk: Uint8Array) {
+    return nacl.sign.keyPair.fromSecretKey(sk);
+  }
+
+export default class utils {
+
+    static createWallet(): object {
+        const wallet = algosdk.generateAccount();
+        const mnemonic = algosdk.secretKeyToMnemonic(wallet.sk);
+        const privateKey = '0x' + Buffer.from(wallet.sk).toString('hex')
+        const keypair = keyPairFromSecretKey(wallet.sk)
+        const publicKey = '0x' + Buffer.from(keypair.publicKey).toString('hex')
+
+        return {
+            mnemonic,
+            privateKey,
+            publicKey,
+            address: wallet.addr
+        }
+    }
+
+    static getWallet(mnemonic: string): object {
+        const wallet = algosdk.mnemonicToSecretKey(mnemonic)
+        const privateKey = '0x' + Buffer.from(wallet.sk).toString('hex')
+        const keypair = keyPairFromSecretKey(wallet.sk)
+        const publicKey = '0x' + Buffer.from(keypair.publicKey).toString('hex')
+
+        return {
+            mnemonic: mnemonic,
+            privateKey,
+            publicKey,
+            address: wallet.addr
+        }
+    }
+
+    static getPublicKey() {
+        throw new Error('Not implemented');
+    }
+
+    static getAddress(privateKey: string): string {
+        const sk = algosdk.secretKeyToMnemonic(privateKeyToArray(privateKey))
+        const wallet = algosdk.mnemonicToSecretKey(sk)
+        return wallet.addr
+    }
+
+    static signTransaction(privateKey: string, transaction: any): object {
```

Comment:
Why is this here? It's not part of the `utils.js` interface.

Reply:
Wouldnt we use this lib to sign transactions as well?

Reply:
That wasn't the original intent of the library, but we could.

What's the benefit?

Reply:
i was looking at the purpose of this library as doing all the "offline" parts, seeing the signMessage method confirmed it for me, but if you see this as something that only creates addresses, that's fair as well and would simplify things but then why is the signMessage stuff in here?

benefit in theory is that we then don't need to call the chain sdks in our code to sign transaction and then just send the signed transaction to the chain.

Reply:
Sure. Let's run with that approach for now and see how it goes. Pretty easy to refactor and remove it later if needed.